frontend/packages/console-app/src/components/detect-perspective/DetectPerspective.tsx

```diff
@@ -7,8 +7,21 @@ type DetectPerspectiveProps = {
   children: React.ReactNode;
 };
 
+const getPerspectiveURLParam = () => {
+  const PerspectiveURLParam = 'perspective';
+  const PerspectiveParamValues = ['dev', 'admin'];
```

Comment:
Hard-coding this list could be problematic since a dynamic plugin can contribute a new perspective we don't know about. The list you have is correct for 4.8, but might not be accurate for 4.9 and later. To correctly get the list, you need to use

```ts
  const perspectiveExtensions = useExtensions<Perspective>(isPerspective);
```

See https://github.com/spadgett/console/blob/b8f030ffe33a59c430a8c618db765cda569fce7c/frontend/public/components/nav/nav-header.tsx#L22

It's potentially a problem when showing the perspectives in ACM as well. I don't have a great answer at this time since getting the complete perspective list requires loading and running a dynamic plugin. Maybe we need something on the `ConsolePlugin` resource that indicates the plugin adds a perspective, which ACM can read.

Reply:
Maybe it's better to have a console backend endpoint that reads the enabled plugin manifests and returns the list of perspectives. (@dtaylor113 not something to worry about for this PR.)

Reply:
Updated to use `const perspectiveExtensions = useExtensions<Perspective>(isPerspective);`